gb-dbg/src/debugger/breakpoints/breakpoint_node.rs

```diff
@@ -0,0 +1,122 @@
+use crate::dbg_interfaces::{CpuRegs, RegisterDebugOperations};
+use anyhow::anyhow;
+use nom::branch::alt;
+use nom::bytes::complete::{tag, take_while_m_n};
+use nom::combinator::map;
+use nom::sequence::tuple;
+use nom::IResult;
+use std::fmt::{Display, Formatter};
+use std::str::FromStr;
+
+#[derive(Debug, Copy, Clone)]
+enum Operator {
+    Eq,
+    NotEq,
+    Sup,
+    Inf,
+    SupEq,
+    InfEq,
+}
+
+impl Display for Operator {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        match self {
+            Operator::Eq => write!(f, "=="),
+            Operator::NotEq => write!(f, "!="),
+            Operator::Sup => write!(f, ">"),
+            Operator::Inf => write!(f, "<"),
+            Operator::SupEq => write!(f, ">="),
+            Operator::InfEq => write!(f, "<="),
+        }
+    }
+}
+
+#[derive(Debug)]
+pub struct BreakpointNode {
+    lhs: CpuRegs,
+    op: Operator,
+    rhs: u16,
+}
+
+impl Display for BreakpointNode {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        if let CpuRegs::PC = self.lhs {
+            if let Operator::Eq = self.op {
+                return write!(f, "0x{:04X}", self.rhs);
+            }
+        }
+
+        write!(f, "{:?} {} 0x{:04X}", self.lhs, self.op, self.rhs)
+    }
+}
+
+impl BreakpointNode {
+    pub fn compute<T: RegisterDebugOperations>(&self, regs: &T) -> bool {
+        match self.op {
+            Operator::Eq => u16::from(regs.cpu_get(self.lhs).unwrap()) == self.rhs,
+            Operator::NotEq => u16::from(regs.cpu_get(self.lhs).unwrap()) != self.rhs,
+            Operator::Sup => u16::from(regs.cpu_get(self.lhs).unwrap()) > self.rhs,
+            Operator::Inf => u16::from(regs.cpu_get(self.lhs).unwrap()) < self.rhs,
+            Operator::SupEq => u16::from(regs.cpu_get(self.lhs).unwrap()) >= self.rhs,
+            Operator::InfEq => u16::from(regs.cpu_get(self.lhs).unwrap()) <= self.rhs,
+        }
+    }
+
+    pub fn new_simple(address: u16) -> Self {
+        Self {
+            lhs: CpuRegs::PC,
+            op: Operator::Eq,
+            rhs: address,
+        }
+    }
+}
+
+impl FromStr for BreakpointNode {
+    type Err = anyhow::Error;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let (rest, (reg, op, val)) = match tuple((register, operator, value))(s) {
+            Ok(ret) => ret,
+            Err(_) => {
+                return Err(anyhow!("Invalid input"));
+            }
```

Comment:
[`map_err`](https://doc.rust-lang.org/std/result/enum.Result.html#method.map_err)

Reply:
I can't get the right syntax to work with the compiler :/